src/components/Counter.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import './Counter.scss';
+
+type State = {
+  number: number,
+};
+
+class Counter extends React.Component<{}, State> {
+  state = {
+    number: 0,
+  };
+
+  addOne = () => {
+    this.setState((currState) => ({ number: currState.number + 1 }));
+  };
+
+  add100 = () => {
+    this.setState((currState) => ({ number: currState.number + 100 }));
+  };
+
+  addOneOr100 = () => {
+    const remainder = this.state.number % 5;
+
+    if (remainder === 0) {
+      this.add100();
+
+      return;
+    }
+
+    this.addOne();
+  };
+
+  render() {
+    return (
+      <div className="Counter">
+        <h1 className="Counter__title">
+          Current number:
+          {this.state.number}
+        </h1>
+        <button type="button" className="Counter__button" onClick={this.addOne}>
+          Add 1
+        </button>
```

Comment:
move props to the new line when they're more than 2 of them

```suggestion
        <button
          type="button"
          className="Counter__button"
          onClick={this.addOne}
        >
          Add 1
        </button>
```